src/psyclone/psyGen.py

```diff
@@ -3276,6 +3276,67 @@ def __init__(self, parent=None, variable_name="", valid_loop_types=None,
         self._variable_name = variable_name
         self._id = ""
 
+    @staticmethod
+    def create(var_name, start, stop, step, children):
+        '''Create a Loop instance given valid instances of a variable name,
+        start, stop and step nodes, and a list of child nodes for the
+        loop body.
+
+        :param str var_name: the PSyIR node containing the variable \
+            name of the loop iterator.
+        :param start: the PSyIR node determining the value for the \
+            start of the loop.
+        :type start: :py:class:`psyclone.psyGen.Node`
+        :param end: the PSyIR node determining the value for the end \
+            of the loop.
+        :type end: :py:class:`psyclone.psyGen.Node`
+        :param step: the PSyIR node determining the value for the loop \
+            step.
+        :type step: :py:class:`psyclone.psyGen.Node`
+        :param children: a list of PSyIR nodes contained in the \
+            loop.
+        :type children: list of :py:class:`psyclone.psyGen.Node`
+
+        :returns: a Loop instance.
+        :rtype: :py:class:`psyclone.psyGen.Loop`
+
+        :raises GenerationError: if the arguments to the create method \
+            are not of the expected type.
+
+        '''
+        if not isinstance(var_name, str):
+            raise GenerationError(
+                "var_name argument in create method of Loop class "
+                "should be a string but found '{0}'."
+                "".format(type(var_name).__name__))
+        for name, instance in [("start", start), ("stop", stop),
+                               ("step", step)]:
+            if not isinstance(instance, Node):
+                raise GenerationError(
+                    "{0} argument in create method of Loop class should "
+                    "be a PSyIR Node but found '{1}'."
+                    "".format(name, type(instance).__name__))
+        if not isinstance(children, list):
+            raise GenerationError(
+                "children argument in create method of Loop class "
+                "should be a list but found '{0}'."
+                "".format(type(children).__name__))
+        for child in children:
+            if not isinstance(child, Node):
+                raise GenerationError(
+                    "child of children argument in create method of Loop "
+                    "class should be a PSyIR Node but found '{0}'."
+                    "".format(type(child).__name__))
+
+        loop = Loop()
+        start.parent = loop
+        stop.parent = loop
+        step.parent = loop
+        schedule = Schedule(parent=loop, children=children)
```

Comment:
You still need to iterate over children to say: `child.parent = schedule` and add a test that this connection is done.

Reply:
Added tests for all child/parent relationships in create methods and all now pass.